android/pytorch_android/CMakeLists.txt

```diff
@@ -79,10 +77,8 @@ else()
   # Prefer dynamic linking on the host
   target_link_libraries(pytorch
       fbjni
-      -Wl,--gc-sections
-      -Wl,--whole-archive
+      -Wl,-dead_strip
       torch
-      -Wl,--no-whole-archive
```

Comment:
Unfortunately, I suspect this breaks the Linux build, which links with gold.  Any way we can get both Mac and Linux builds to work?

Also, this is not related to the subject of the diff.  Please use separate diffs for unrelated changes.